adapters/grid/grid.go

```diff
@@ -15,11 +15,50 @@ type GridAdapter struct {
 	endpoint string
 }
 
+func processImp(imp *openrtb.Imp) error {
+	// get the grid extension
+	var ext adapters.ExtImpBidder
+	var gridExt openrtb_ext.ExtImpGrid
+	if err := json.Unmarshal(imp.Ext, &ext); err != nil {
+		return err
+	}
+	if err := json.Unmarshal(ext.Bidder, &gridExt); err != nil {
+		return err
+	}
+	if gridExt.Uid == 0 {
+		err := &errortypes.BadInput{
+			Message: "uid is empty",
+		}
+		return err
+	}
+	// no error
+	return nil
+}
+
 // MakeRequests makes the HTTP requests which should be made to fetch bids.
 func (a *GridAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
 	var errors = make([]error, 0)
 
-	reqJSON, err := json.Marshal(request)
+	// copy the request, because we are going to mutate it
+	requestCopy := *request
+	// this will contain all the valid impressions
+	var validImps []openrtb.Imp
+	// pre-process the imps
+	for _, imp := range requestCopy.Imp {
+		if err := processImp(&imp); err == nil {
+			validImps = append(validImps, imp)
+		} else {
+			errors = append(errors, err)
+		}
+	}
+	if len(validImps) == 0 {
+		err := fmt.Errorf("No valid impressions for grid")
```

Comment:
Please add a test (_test.go or json based) for this condition and verify the `uid is empty` input error.

Reply:
Hello, we made the necessary updates, please review